gusto/diagnostics.py

```diff
@@ -39,6 +69,10 @@ def compute(self, state):
         """ Compute the diagnostic field from the current state"""
         pass
 
+    def _setup_solver(self, state):
+        """Setup solvers for calculating this diagnostic field"""
+        pass
```

Comment:
Not ideal, but for now why can't the compute method just call the setup_solver method and have it set up the solver on the first call, or return the solver otherwise - just like the current field method does? Otherwise it is a bit confusing to have a method in the base DiagnosticField class that most diagnostic fields don't need.